source/source.go

```diff
@@ -250,12 +252,23 @@ func endpointsForHostname(hostname string, targets endpoint.Targets, ttl endpoin
 	var endpoints []*endpoint.Endpoint
 
 	var aTargets endpoint.Targets
+	var aaaaTargets endpoint.Targets
 	var cnameTargets endpoint.Targets
 
 	for _, t := range targets {
 		switch suitableType(t) {
 		case endpoint.RecordTypeA:
-			aTargets = append(aTargets, t)
+			if !isIPv6String(t) {
```

Comment:
Please keep the good on the left and errors/specials on the right:
```
if isIPv6String(t) {
    continue
}
aTargets = append(aTargets, t)
```

Same below for AAAA handling